src/SFA.DAS.EmployerApprenticeshipsService.Domain/Models/TransferConnections/TransferConnectionInvitation.cs

```diff
@@ -90,18 +98,32 @@ public void Approve(Data.Entities.Account.Account approverAccount, User approver
             });
         }
 
-        public void Delete(Data.Entities.Account.Account deleterAccount, User deleterUser)
+        public void Delete(Data.Entities.Account.Account deleterAccount, User deleterUser, DeleteTransferConnectionInvitationMode deletionMode)
         {
-            RequiresDeleterAccountIsTheSenderAccount(deleterAccount);
             RequiresTransferConnectionInvitationIsRejected();
 
             var now = DateTime.UtcNow;
+            var deletedBySender = deletionMode.HasFlag(DeleteTransferConnectionInvitationMode.SenderIsDeleting)? true: (bool?) null;
```

Comment:
Could the enum be dropped and the logic simplified to something along these lines?
```
if (deleterAccount.Id == ReceiverAccount.Id)
{
    RequiresNotAlreadyDeletedByReceiver();

    DeletedByReceiver = true;
}
else if (deleterAccount.Id != SenderAccount.Id)
{
    RequiresNotAlreadyDeletedBySender();

    DeletedBySender = true;
}
else
{
    throw new Exception("Requires deleter account is the receiver or sender account.");
}
```